rapidsai/dask-build-environment
# reduce config duplication, update third-party actions

Closes #100 

Follow-up to #97 

Contributes to https://github.com/rapidsai/build-planning/issues/40

Proposes the following changes:

* ~adding Python 3.12 images~
  - *~(now that those have been added upstream in https://github.com/rapidsai/ci-imgs/pull/137)~*
* removing as many sources of duplicated configuration as possible
  - *(with the goal of only needing to update `ci/axis/dask.yaml` in the future for changes like "support a new Python version")*
* updating all third-party GitHub Actions to their latest versions
  - *(which should resolve all the warnings you can see in job summaries, e.g. https://github.com/rapidsai/dask-build-environment/actions/runs/10293483059)
* adds a minimal `.gitignore` and `README.md`

## Notes for Reviewers

### How do I test this?

I can't figure out where these image builds actually happen. At https://github.com/rapidsai/dask-build-environment/actions, I only see the "Check for gpuCI Updates" job running.

Could someone please tell me where `docker build` based on these `Dockerfiles` is run and how I could test these changes?
